### PR TITLE
Handle bad recorddata

### PR DESCRIPTION
**Pull request by rsinger**
The problem that #3 was intended to solve introduced a different - the _actual issue_ #3 was trying to overcome was the fact that the target SRU server was returning invalid recordData responses.

Unfortunately, that change introduced a backwards incompatible change here.  And this will be sort of a backwards incompatible change _back_ to the old behavior, but it comes close to supporting the use case #3 was trying to solve.

It also introduces linting, cuts off PHP 5.3 support, and adds tests.

### CODE REVIEW

**Review by timhodson — APPROVED** [review]
This looks good to me. we'll have to try it with TARL and see what needs updating there too.

**Comment by rsinger** [comment]
It will definitely be a breaking change in TARL for non-Vubis customers (and probably for them as well), but I don't think it will be a difficult fix.

File: src/SRU/ScanResponse.php [hunk 1]
```diff
@@ -5,14 +5,16 @@ class ScanResponse
...
+    public function __construct($xml)
     {
-        if(is_string($xml))
-        {
+        if (is_string($xml)) {
             $this->doc = new \DOMDocument();
             $this->doc->loadXML($xml);
-        } elseif (is_a($xml, '\DOMDocument'))
-        {
+        } elseif (is_a($xml, '\DOMDocument')) {
```
Comment (rgubby) [inline src/SRU/ScanResponse.php:17]: No need to change this as it's a linting change, but we would usually do `instanceof` rather than `is_a`

**Review by rgubby — APPROVED** [review]
This all looks fine to me.

**Comment by rsinger** [comment]
TARL build passed with no changes needed!

**Pull request merged**